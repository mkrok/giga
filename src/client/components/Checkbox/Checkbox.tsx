import { useEffect, useState } from 'react';
import styled from 'styled-components';
import { useAppData } from '~/client/context/AppStateProvider';
import Data from '../Data/Data';

type CheckboxProps = {
  index: number;
};

const CheckboxStyle = styled.div`
  font-size: 1.5rem;
  width: 2rem;
  height: 2rem;
`;

const Checkbox = ({ index }: CheckboxProps) => {
  const [isChecked, setIsChecked] = useState(false);
  const { appData, setAppData } = useAppData();
  const switches = appData.checkboxes;
  useEffect(() => {
    switches[index] = isChecked;
    setAppData({
      ...appData,
      checkboxes: switches,
    });
  }, [isChecked]);
  return (
    <CheckboxStyle>
      <label>{`${index}`}</label>
      <input
        checked={isChecked}
        type="checkbox"
        onChange={() => {
          setIsChecked(prev => !prev);
        }}
      />
    </CheckboxStyle>
  );
};
export default Checkbox;
