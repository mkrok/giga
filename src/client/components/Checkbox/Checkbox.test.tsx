import renderer from 'react-test-renderer';
import Checkbox from '.';

describe('Checkbox', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Checkbox index={0} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
