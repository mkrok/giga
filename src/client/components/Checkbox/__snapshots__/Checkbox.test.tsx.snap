// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<div
  className="sc-beqWaB fvYjOV"
>
  <label>
    0
  </label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
